app/assets/javascripts/application.js

```diff
@@ -3,7 +3,7 @@
 //= require govuk_publishing_components/lib/trigger-event
 //= require govuk_publishing_components/lib/cookie-functions
 
-//= require analytics
+//= require load-analytics
```

Comment:
This looks well thought out but I've only given it a quick glance. My initial question is did you consider moving some or all of this into `govuk_publishing_components`? We're aiming to retire `static` eventually and putting this together with the rest of the analytics code seems like a good idea. Fine if there's a good reason why not, I'll find some time to do a proper review soon...

Reply:
@andysellick Nope I didn't consider `govuk_publishing_components`... but yeah I guess the only reason it was in `static` was because that was where the environment variables lived which we don't need anymore. 

Should we pause reviewing this for now and i'll move it into `govuk_publishing_components`?

Reply:
I think it would be worth investigating, yes. As I said, there might be a reason to keep it in `static` (or at least part of it) but worth checking.

Reply:
Thanks

Reply:
I've had another thought overnight - it might also be worth splitting out this code to separate the UA stuff and the GA4 stuff - maybe have one init script in the existing `analytics/` dir and one in the `analytics-ga4/` dir. Again, I've not looked into the code to check the feasibility of this, so it's not essential if it proves tricky.